services/worker-manager/src/providers/provider.js

```diff
@@ -84,6 +84,46 @@ class Provider {
     throw new ApiError('not supported for this provider');
   }
 
+  /**
+   * Workers that are being provisioned by worker manager are expected to:
+   * 1. Start (instance is running)
+   * 2. Register (worker is registered with worker manager)
+   * 3. Do work (call queue.claimWork/queue.reclaimTask)
+   *
+   * If worker does not register within given timeout, it will be removed after `terminateAfter` time.
+   * If worker fails to call queue.claimWork, `queue_worker.first_claim` would be set to null.
+   * If worker does not call queue.reclaimTask or stops calling queue.claimWork,
+   * `queue_worker.last_date_active` would not be updated.
+   *
+   * Workers that are registered, but don't have `first_claim`
+   * or `last_date_active` is older than inactivity timeout are considered to be zombies.
+   */
+  static isZombie({ worker, activityTimeout, createdTimeout }) {
+    activityTimeout = activityTimeout || 1000 * 60 * 60 * 2; // last active within 2 hours
+    createdTimeout = createdTimeout || 1000 * 60 * 60 * 1; // created at least 1 hour ago
```

Comment:
These should probably be values specified in the `lifecycle` property of the worker pool, rather than hard-coded. As mentioned in the meeting today, they can probably be much smaller for AWS and Google.

Reply:
Thanks Dustin,  yes, I moved them into the `lifecycle` and left a single `claimTimeout` parameter